chain/chain/src/chain.rs

```diff
@@ -1145,11 +1241,10 @@ impl Chain {
                         let tail_height = self.store.tail()?;
                         // we only add blocks that couldn't have been gc'ed to the orphan pool.
                         if block_height >= tail_height {
-                            let block_hash = *block.hash();
-                            let orphan = Orphan { block, provenance, added: Clock::instant() };
-
-                            self.orphans.add(orphan);
-
+                            let block_hash = block.hash();
+                            self.orphans.add(
```

Comment:
I know the code was not written by you, but I think you can call `self.save_orphan` here. This way, it's clear that `self.save_orphan` is the function to call when to add orphan to the orphan pool, and because block is validated there, we guarantee that all blocks in the orphan pool have been validated.

Reply:
I ended up leaving it as is for now to minimise diff.